src/journal/journald-server.h

```diff
@@ -60,6 +60,13 @@ typedef struct JournalStorage {
         JournalStorageSpace space;
 } JournalStorage;
 
+/* This structure will be kept in $RUNTIME_DIRECTORY/seqnum and is mapped by journald, and is used to
+ * maintain the sequence number counter with its seqnum ID */
+typedef struct SeqnumData {
+        sd_id128_t id;
```

Comment:
given this is supposed to survive restarts, and thus cross-version up/downgrade, is it worth adding a version field, just in case it will be needed?

Reply:
so i purposefully kept this separate from anything else, gave it its own little file, in /run, so that it's isolated enough to not be affected by compat breakages. It's a very encapsulated concept: the sequence number and its immediately associated ID. these things belong together like laurel & hardy, you can't really separate them.

or to say this differently: in the very very unlikely event we need to extend this we have two ways out: the two existing fields are fixed size, hence we can just add another field to the end (which could even be a version filed if we wanted to), and then simply use the size of this file to determine the version. or we could just add anothe file like this next to it, with the new data.